addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportApi.java

```diff
@@ -0,0 +1,318 @@
+                            throw new ApiException(Type.ILLEGAL_PARAMETER, PARAM_INC_RISKS);
+                        }
+                        reportData.setIncludeRisk(riskIndex, true);
+                    }
+                } else {
+                    reportData.setIncludeAllRisks(true);
+                }
+
+                // Alerts
+                reportData.setAlertTreeRootNode(extReports.getFilteredAlertTree(reportData));
+
+                try {
+                    extReports.generateReport(reportData, template, reportFilePath, display);
+                } catch (Exception e) {
+                    throw new ApiException(Type.INTERNAL_ERROR, e);
+                }
+                return new ApiResponseElement("generatedReportPath", reportFilePath);
+            default:
+                throw new ApiException(Type.BAD_ACTION);
+        }
+    }
+
+    @Override
+    public ApiResponse handleApiView(String name, JSONObject params) throws ApiException {
+        LOG.debug("Request for handleApiAction: {} (params: {})", name, params);
+        switch (name) {
+            case VIEW_TEMPLATES:
+                ApiResponseList resultList = new ApiResponseList(name);
+                extReports
+                        .getTemplates()
+                        .forEach(
+                                t ->
+                                        resultList.addItem(
+                                                new ApiResponseElement(
+                                                        "template", t.getConfigName())));
+                return resultList;
+            case VIEW_TEMPLATE_DETAILS:
+                Template template =
+                        extReports.getTemplateByConfigName(params.getString(PARAM_TEMPLATE));
+                if (template == null) {
+                    throw new ApiException(
+                            Type.DOES_NOT_EXIST,
+                            Constant.messages.getString(
+                                    "reports.api.error.templateDoesNotExist",
+                                    params.getString(PARAM_TEMPLATE)));
+                }
+                ApiResponseSet<Object> resultSet = new ApiResponseSet<>(name, new HashMap<>());
+                resultSet.put("name", template.getDisplayName());
+                resultSet.put("format", template.getFormat());
+                JSONArray sections = new JSONArray();
+                sections.addAll(template.getSections());
+                resultSet.put("sections", sections);
+                JSONArray themes = new JSONArray();
+                themes.addAll(template.getThemes());
+                resultSet.put("themes", themes);
+                return resultSet;
+            default:
+                throw new ApiException(Type.BAD_VIEW);
+        }
+    }
+
+    private static boolean isContainedParam(String paramName, JSONObject params) {
+        return params.containsKey(paramName) && !params.getString(paramName).isEmpty();
```

Comment:
Does this need to handle empty or not contained situations? (Throwing an API Exception)

Reply:
No, it's used only for checking if optional param values are present or not. So it's not an exception if they're not contained or empty (because they're optional).

Reply:
Thanks, sorry I missed that.